From 779ab83d86b23be38254cd98c2bc9e7ffa84352f Mon Sep 17 00:00:00 2001
From: Semih Yavuz <semih.yavuz@qt.io>
Date: Wed, 24 Sep 2025 18:05:39 +0200
Subject: [PATCH 095/239] qmlformat: fix incorrect indent

Lexer tokenizes some type names as T_RESERVED_WORD.  When it doesn't
break here, state machine hits the default case where it  leaves
the type annotation state but token is not advanced. This
causes to state machine corruption where all subsequent tokens
are processed under incorrect states, resulting in wrong indentation.

f649231204f8e864ed1ef88257bd88eaca06731c fixes this issue on dev.

Pick-to: 6.10.0 6.9 6.8
Fixes: QTBUG-138709
Change-Id: I9956ee07c36067abdba903897e5463cbd28fcdd9
Reviewed-by: Sami Shalayel <sami.shalayel@qt.io>
---
 src/qmldom/qqmldomcodeformatter.cpp                            | 1 +
 tests/auto/qmldom/domdata/reformatter/typeAnnotations.qml      | 2 ++
 .../qmldom/domdata/reformatter/typeAnnotationsReformatted.qml  | 3 +++
 3 files changed, 6 insertions(+)

diff --git a/src/qmldom/qqmldomcodeformatter.cpp b/src/qmldom/qqmldomcodeformatter.cpp
index 017cef80b5..e51d3875d0 100644
--- a/src/qmldom/qqmldomcodeformatter.cpp
+++ b/src/qmldom/qqmldomcodeformatter.cpp
@@ -626,6 +626,7 @@ void FormatPartialStatus::handleTokens()
 
         case StateType::TypeAnnotation:
             switch (kind) {
+            case QQmlJSGrammar::T_RESERVED_WORD:
             case QQmlJSGrammar::T_IDENTIFIER:
             case QQmlJSGrammar::T_DOT:
                 break;
diff --git a/tests/auto/qmldom/domdata/reformatter/typeAnnotations.qml b/tests/auto/qmldom/domdata/reformatter/typeAnnotations.qml
index 63c4f5471c..a72b43d9e6 100644
--- a/tests/auto/qmldom/domdata/reformatter/typeAnnotations.qml
+++ b/tests/auto/qmldom/domdata/reformatter/typeAnnotations.qml
@@ -9,4 +9,6 @@ var w = object.width / 3
 console.debug(w)
 return w
 }
+
+function test() : int {return 53;}
 }
diff --git a/tests/auto/qmldom/domdata/reformatter/typeAnnotationsReformatted.qml b/tests/auto/qmldom/domdata/reformatter/typeAnnotationsReformatted.qml
index 3b304176fb..f19e44c1e7 100644
--- a/tests/auto/qmldom/domdata/reformatter/typeAnnotationsReformatted.qml
+++ b/tests/auto/qmldom/domdata/reformatter/typeAnnotationsReformatted.qml
@@ -9,4 +9,7 @@ Item {
         console.debug(w);
         return w;
     }
+    function test(): int {
+        return 53;
+    }
 }
-- 
2.51.2

